docker/entrypoint.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+    local var="$1"
+    local fileVar="${var}_FILE"
+    local def="${2:-}"
+    if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+        echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+        exit 1
+    fi
+    local val="$def"
+    if [ "${!var:-}" ]; then
+        val="${!var}"
+    elif [ "${!fileVar:-}" ]; then
+        val="$(< "${!fileVar}")"
+    fi
+    export "$var"="$val"
+    unset "$fileVar"
+}
+
+file_env 'IOTA_AUTH_USER'
+file_env 'IOTA_AUTH_PASSWORD'
+
+if [[  -z "$IOTA_AUTH_ENABLED" ]]; then
+ echo "***********************************************"
+ echo "WARNING: It is recommended to enable authentication for secure connection"
+ echo "***********************************************"
+else
+    if  [[ -z "$IOTA_AUTH_USER" ]] || [ -z "$IOTA_AUTH_PASSWORD" ]]; then
+        echo "***********************************************"
+        echo "WARNING: Default IoT Agent Auth credentials have not been overridden"
+        echo "The connection to the Context broker fails"
```

Comment:
"The connection to the Context broker fails" -> "The connection to the Context broker will fail" (unsure, but I think the later is more correct)

Reply:
Personally I'd just remove that **second** line, because there is at least one circumstance where it wouldn't fail. Imagine this scenario (I know these are just edge cases, but...):

1) Override `IOTA_AUTH_ENABLED = true`  ✅ 
2) Replace the default `config.js` with an attached `VOLUME` holding the *real* username and /or password in plain text. 😮 

It's still not really **safe** IMHO, but it would work and therefore the connection wouldn't fail so the second line of the statement would be incorrect.

Maybe someone is holding their username/passwords as _configuration as code_ passing an encrypted file onto a machine and then mounting in within a docker instance?

Reply:
What about

"The connection to the Context broker most probably fail"

?

Removing the line is also ok from my point of view.

Reply:
Fixed with https://github.com/telefonicaid/iotagent-json/pull/380/commits/3a5d1c5f5d74cc9b34444041157ce0f03f3ef7f0

Reply:
(Actually it was fixed in commit 04ecf9c897835733a9715527617f84f1a790343c)